src/ert/_c_wrappers/enkf/config/gen_kw_config.py

```diff
@@ -146,3 +154,51 @@ def get_priors(self) -> List["PriorDict"]:
                 }
             )
         return priors
+
+    def transform(self, index: int, value: float) -> float:
+        return self._transform(index, value)
+
+    @staticmethod
+    def values_from_files(
+        realizations: List[int], name_format: str, keys: List[str]
+    ) -> npt.NDArray[np.double]:
+        df_values = pd.DataFrame()
+        for iens in realizations:
+            df = pd.read_csv(
+                name_format % iens,
+                delim_whitespace=True,
+                header=None,
+            )
+            # This means we have a key: value mapping in the
+            # file otherwise it is just a list of values
+            if df.shape[1] == 2:
+                # We need to sort the user input keys by the
+                # internal order of sub-parameters:
+                df = df.set_index(df.columns[0])
+                values = df.reindex(keys).values.flatten()
+            else:
+                values = df.values.flatten()
+            df_values[f"{iens}"] = values
+        return df_values.values
+
+    @staticmethod
+    def sample_values(
+        parameter_group_name: str,
+        keys: List[str],
+        global_seed: str,
+        active_realizations: List[int],
+        nr_samples: int,
+    ) -> npt.NDArray[np.double]:
+        parameter_values = []
+        for key in keys:
+            key_hash = sha256(
+                global_seed.encode("utf-8")
+                + f"{parameter_group_name}:{key}".encode("utf-8")
+            )
+            seed = np.frombuffer(key_hash.digest(), dtype="uint32")
```

Comment:
Is it correct that the idea of `key_hash` here is to get the same seed for a specific `global_seed` and a specific parameyer/key pair?
What is the idea behind using `np.frombuffer`? AFAIK it returns an array and I'm not sure why an array is needed as a seed here.

Reply:
This is the behavior currently on main: https://github.com/equinor/ert/blob/b828844409c74bcd22bd57f7196c9861a4bb8694/src/ert/_c_wrappers/enkf/data/gen_kw.py#L144 

Probably worth a discussion :shrug: